tests.py

```diff
@@ -1917,6 +1917,193 @@ def test_basic_auth_second_route_same_path_respected(self):
         self.assertEqual(response.data, b"ok")
 
 
+class SharedTokenTestCase(unittest.TestCase):
+    def get_shared_token_response(self, custom_headers=None):
+        if custom_headers == None:
+            custom_headers = {"x-cdn-secret": "my-secret"}
+
+        headers = {
+            "x-cf-forwarded-url": "http://somehost.com/",
+            "x-forwarded-for": "1.2.3.4, 1.1.1.1, 1.1.1.1",
+        } | custom_headers
+
+        return urllib3.PoolManager().request(
+            "GET",
+            url="http://127.0.0.1:8080/",
+            headers=headers,
+        )
+
+    @parameterized.expand(
+        [({}, 403), ({"x-cdn-secret": "not-my-secret"}, 403), (None, 200)]
+    )
```

Comment:
Might just be me, but I think it's easier to grok with one test case per line...
```suggestion
    @parameterized.expand([
        ({}, 403),
        ({"x-cdn-secret": "not-my-secret"}, 403),
        (None, 200),
    ])
```